### PR TITLE
MINOR: Remove waitUntilTrue from TestUtils.scala.

### PR DESCRIPTION
**Pull request by Wuzhengyu97**
Removed the waitUntilTrue function from TestUtils.scala and replaced it with waitForCondition.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/test/TestUtils.java [hunk 1]
```diff
@@ -393,6 +397,16 @@ public static void waitForCondition(final TestCondition testCondition, final lon
...
 
+    public static void waitForCondition(
+            final TestCondition testCondition,
+            String conditionDetails,
+            final long maxWaitMs,
+            final long pollIntervalMs
+    ) throws InterruptedException {
+        waitForCondition(testCondition, maxWaitMs, pollIntervalMs, () -> conditionDetails);
+    }
+
```
Comment (dengziming) [inline clients/src/test/java/org/apache/kafka/test/TestUtils.java]: nit: redundant inline

File: clients/src/test/java/org/apache/kafka/test/TestUtils.java [hunk 2]
```diff
@@ -383,6 +383,10 @@ public static void waitForCondition(final TestCondition testCondition, final lon
         waitForCondition(testCondition, maxWaitMs, () -> conditionDetails);
     }
 
+    public static void waitForCondition(final TestCondition testCondition, String conditionDetails, final long maxWaitMs) throws InterruptedException {
```
Comment (dengziming) [inline clients/src/test/java/org/apache/kafka/test/TestUtils.java]: It's not recommend to keep 2 methods with same parameters but different order  for the simple reason just to reduce code change, especially in test module. so how about remove it?

File: clients/src/test/java/org/apache/kafka/test/TestUtils.java [hunk 3]
```diff
@@ -393,6 +397,16 @@ public static void waitForCondition(final TestCondition testCondition, final lon
         waitForCondition(testCondition, maxWaitMs, DEFAULT_POLL_INTERVAL_MS, conditionDetailsSupplier);
     }
 
+    public static void waitForCondition(
```
Comment (dengziming) [inline clients/src/test/java/org/apache/kafka/test/TestUtils.java]: this method is only used 8 times, we can also remove it.

**Comment by dengziming** [comment]
LGTM, except some improvement suggestions.

**Review by dengziming — CHANGES_REQUESTED** [review]
Thanks for this clean up, left some improvement suggestions.

**Comment by Wuzhengyu97** [comment]
@dengziming 
Thanks for the heads up, I've fixed the problem you mentioned.

**Pull request closed without merging**